### ISSUE TITLE
[Swift4] OpenAPI V3.0 - Server Tags convert into injectable objects?

### ISSUE THREAD

**Issue by inforeqd512** [swagger-api/swagger-codegen#6802]
##### Description

This may be an enhancement request for when you have a capability to codegen v3.0 of the Swagger specs. 

In V3.0 there is a servers tag where we can specify multiple servers eg production, staging, development. If these can be converted into injectable objects, then we can use it to instantiate requests to appropriate base paths depending on the destination. 

Currently, the Open API specs just gives a URL and a description tags for each of the server. Neither can be used to create a name for the object. So may need to work with OpenAPi to add a 'name' tag? Or if you can come up with another way?

##### Swagger-codegen version

NA

##### Swagger declaration file content or url

NA

##### Command line used for generation

NA

##### Steps to reproduce

NA

##### Related issues/PRs

NA

##### Suggest a fix/enhancement

NA

**Comment by wing328** [comment]
@inforeqd512 thanks for the feedback.

For other API clients (e.g. Ruby, Python, PHP), the base path (e.g. https://petstore.swagger.io/v1) can be changed during runtime. Maybe the Swift client can do something similar as a workaround for the time being.

cc @jgavris @ehyche @Edubits @jaz-ah

**Comment by inforeqd512** [comment]
Yes thank you.. We are doing that for now, but were hoping to be able to link it instead with end points in the spec
